File: Programmers/level3/네트워크.py
#dfs
def solution(n, computers):
    answer = 0
    graph=[[] for _ in range(n)]
    visited=[False]*n
    print(graph,visited)
    for i in range(n):
        for j in range(n):
            if i!=j and computers[i][j]==1:
                graph[i].append(j)
    def dfs(graph,v,visited):
        visited[v]=True
        for i in graph[v]:
            if not visited[i]:
                dfs(graph,i,visited)

    for i in range(n):
        if visited[i]==False:
            dfs(graph,i,visited)
            answer+=1
        
    return answer

File: Programmers/level3/네트워크.js
function solution(n, computers) {
  var answer = 0;
  let graph = [];
  let visited = [];
  let len = computers.length;
  for (let i = 0; i <= len; i++) {
    graph[i] = [];
    visited[i] = false;
  }
  for (let i = 0; i < len; i++) {
    for (let j = 0; j < len; j++) {
      if (i != j && computers[i][j] === 1) {
        graph[i + 1].push(j + 1);
      }
    }
  }
  //연결 그래프 만들기

  function dfs(graph, v, visited) {
    visited[v] = true;
    for (let i of graph[v]) {
      if (!visited[i]) {
        dfs(graph, i, visited);
      }
    }
  }
  //dfs돌리기

  for (let i = 1; i <= len; i++) {
    if (visited[i] === false) {
      dfs(graph, i, visited);
      answer += 1;
    }
  }
  return answer;
}
